### ISSUE TITLE
토큰 검증 및 accessToken 재발급

### ISSUE THREAD

**Issue by Jeonwoojae** [whale2023/job-recommendation-backend#5]
api gateway에서 토큰 검증을 수행하고 access token이 만료되었다면 refresh token을 사용하여 재발급.

방법 1. Token 검증 과정에서 만료 시 클라이언트에 만료 오류를 보내고 클라이언트는 재발급 api로 요청하고 받은 토큰으로 다시 기존 요청 통신

방법 2. 서버에서 만료 시 바로 refresh token으로 재발급 하고 요청의 결과값과 access token을 포함하여 전송.

방법 2를 하고싶은데 응답 값이 변할 수 있다는게 불편함

**Comment by agfalcon** [comment]
방법 1. 사용자가 access token으로 로그인 시 만료되었을 때 오류 보내고 그럼 사용자가 refresh token을 보내면 서버에서 access token을 만들어서 다시 보내주는 방법

방법 2는 이해가 잘 안됩니다 ㅠㅠ

방법 1 제가 이해한게 맞나요???

**Comment by Jeonwoojae** [comment]
네 이해하신게 맞는데 안드로이드에서 사용자가 다시 token을 보내는 것을 사용자는 인식 하지 못하게 가능할까요?

방법 2는 만료된 access token을 받으면 서버에서 바로 재발급 해 기존에 요청한 응답과 함께 access token을 주는겁니다. 그래서 기본적인 서버 응답에는 토큰이 있을 수도, 없을 수도 있게 되는 것이 불편합니다.

**Comment by agfalcon** [comment]
2번은 다시 refresh token을 보내는게 비효율적이라서 refresh token을 생략하고 바로 access token을 받는 걸로 이해했습니다.
이때는 로그인이 성공이 되는 것이라면 만료가 되든 안되든 서버에서 사용자에게 지금 유효한 access token을 보내주는 거는 조금 비효율적일까요??
